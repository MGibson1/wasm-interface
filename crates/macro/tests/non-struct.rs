use wasm_interfacegen_macro::JsInterface;


#[derive(JsInterface)]
enum OnAnEnum{}

#[derive(JsInterface)]
union OnAUnion {
    field: i8,
}

pub fn main(){}
