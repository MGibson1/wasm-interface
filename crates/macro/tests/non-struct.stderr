error: JsInterface only supports structs
 --> tests/non-struct.rs:5:1
  |
5 | enum OnAnEnum{}
  | ^^^^

error: JsInterface only supports structs
 --> tests/non-struct.rs:8:1
  |
8 | union OnAUnion {
  | ^^^^^
